packages/core/http/core-http-server/src/versioning/types.ts

```diff
@@ -203,8 +203,44 @@ export interface VersionedRouter<Ctx extends RqCtx = RqCtx> {
 export type VersionedRouteRequestValidation<P, Q, B> = RouteValidatorFullConfigRequest<P, Q, B>;
 
 /** @public */
+export interface VersionedCustomRouteValidation {
+  custom: RouteValidationFunction<unknown>;
+}
```

Comment:
We already allowed passing in a function... This is a bit awkward but seemed like the simplest way to distinguish custom validation from lazy validation. Also this is not very commonly used.